zerver/tests/test_message_send.py

```diff
@@ -2030,10 +2036,68 @@ def test_private_message_policy(self) -> None:
         with self.assertRaises(JsonableError):
             self.send_personal_message(user_profile, self.example_user("cordelia"))
 
-        bot_profile = self.create_test_bot("testbot", user_profile)
-        self.send_personal_message(user_profile, get_system_bot(settings.NOTIFICATION_BOT))
-        self.send_personal_message(user_profile, bot_profile)
-        self.send_personal_message(bot_profile, user_profile)
+        self.can_send_pm_to_bot(user_profile)
+
+    def test_private_message_policy_admin_only(self) -> None:
+        """
+        Tests that PRIVATE_MESSAGE_POLICY_ADMIN_ONLY works correctly.
+        """
+        test_user_dict = {
+            "admin_user": self.example_user("iago"),
+            "non_admin_user": self.example_user("hamlet"),
+            "other_user": self.example_user("othello"),
+        }
+        self.login_user(test_user_dict["non_admin_user"])
+
+        do_set_realm_property(
+            test_user_dict["non_admin_user"].realm,
+            "private_message_policy",
+            Realm.PRIVATE_MESSAGE_POLICY_ADMIN_ONLY,
+            acting_user=None,
+        )
+        self.can_send_pm_to_bot(test_user_dict["non_admin_user"])
+
+        for user in test_user_dict.values():
+            do_set_realm_property(
+                user.realm,
+                "private_message_policy",
+                Realm.PRIVATE_MESSAGE_POLICY_ADMIN_ONLY,
+                acting_user=None,
+            )
+            with self.assertRaises(JsonableError) as cm:
+                self.send_personal_message(self.example_user("polonius"), user)
+            self.assertEqual(
+                str(cm.exception),
+                "Sending private messages is disabled for guests in this organization.",
+            )
```

Comment:
We can simply use `self.assertRaisesRegex(JsonableError, "Sending private messages is disabled for guests in this organization.")` here instead of checking message separately.

Reply:
Ok